src/components/dashboard/FaceVerification/hooks/useFaceTecVerification.js

```diff
@@ -54,14 +53,17 @@ export default (options = null) => {
     // same like in the props for avoid code modifications
     const [onUIReady, onCaptureDone, onRetry, onComplete, onError, getMaxRetries] = accessors
 
+    const isEmu = await isEmulator
+
     // if cypress is running
-    if (isE2ERunning || isMobileNative) {
+    if (isE2ERunning || isEmu || isMobileNative) {
+      log.debug('skipping fv ui for non real devices', { isMobileNative, isE2ERunning, isEmu })
       try {
         // don't start session, just call enroll with fake data
         // to whitelist user on server
-        await api.performFaceVerification({
-          sessionId: uuid(),
-          enrollmentIdentifier,
+        const sessionId = await api.issueSessionToken()
+        await api.performFaceVerification(enrollmentIdentifier, {
+          sessionId,
```

Comment:
this won't help - as server check for the custom User-Agent Cypress sets in the headers.
So even the sessionId won't be fake, it anyway will call to the zoom's backend and we'll got failed liveness
or unexpected exception because we're sending also fake faceScan

The solution is:
1. leave payload as it was before, with the fake session id
```suggestion
         await api.performFaceVerification({
           sessionId: uuid(),
           enrollmentIdentifier,
```
2. append api.performFaceVerification call with the axios config object containing custom headers:
```javascript
// ...
  lowQualityAuditTrailImage: emptyBase64,
}, { headers: { 'X-Device-Emulated': true } })
```
3. update `src/server/cypress/cypress-middleware.js` on GoodServer:
```javascript
export const cypressMiddleware = (request, _, callNextMiddleware) => {
  const { headers } = request
  const userAgent = headers['user-agent']
  const isCypress = userAgent && userAgent.includes('Cypress')
  const isEmulator = 'X-Device-Emulated' in headers
  const isDevEnv =  'development' === config.env

  assign(request, {
    isEmulator,
    isCypress,
    isE2ERunning: isCypress && isDevEnv,
    isEmulatorRunning: isEmulator && isDevEnv
  })

  callNextMiddleware()
}
```
4. update `src/server/verification/verificationAPI.js`  on GoodServer:
```javascript
const { user, log, params, body: payload, isE2ERunning, isEmulatorRunning } = req

// ...

        // if FV disabled or dups allowed or running Cypress, we're skipping enrollment logic
        if (disableFaceVerification || allowDuplicatedFaceRecords || isE2ERunning || isEmulatorRunning) {
```

Reply:
on dev env we do not verify the face on server. so it works just as expected.
lets save more time, there's no need to give full solution until we decide a change is actually needed.
you can keep your code review more succint

Reply:
@sirpy okay but why do you need  the real session id ? it would work then with fake session id too